modDbImpl/src/org/aion/db/generic/LockedDatabase.java

```diff
@@ -0,0 +1,348 @@
+
+        boolean open = false;
+
+        try {
+            open = database.open();
+        } finally {
+            // releasing write lock
+            lock.writeLock().unlock();
+            return open;
+        }
+    }
+
+    @Override
+    public void close() {
+        // acquire write lock
+        lock.writeLock().lock();
+
+        try {
+            database.close();
+        } finally {
+            // releasing write lock
+            lock.writeLock().unlock();
+        }
+    }
+
+    @Override
+    public boolean commit() {
+        // acquire write lock
+        lock.writeLock().lock();
+
+        boolean success = false;
+
+        try {
+            success = database.commit();
+        } catch (Exception e) {
+            if (e instanceof RuntimeException) {
+                throw e;
+            } else {
+                LOG.error("Could not return empty status due to ", e);
+            }
+        } finally {
+            // releasing write lock
+            lock.writeLock().unlock();
+        }
+        return success;
+    }
+
+    @Override
+    public void compact() {
+        // acquire write lock
+        lock.writeLock().lock();
+
+        try {
+            database.compact();
+        } finally {
+            // releasing write lock
+            lock.writeLock().unlock();
+        }
+    }
+
+    @Override
+    public Optional<String> getName() {
+        // no locks because the name never changes
+        return database.getName();
+    }
+
+    @Override
+    public Optional<String> getPath() {
+        // no locks because the path never changes
+        return database.getPath();
+    }
+
+    @Override
+    public boolean isOpen() {
+        // acquire read lock
+        lock.readLock().lock();
+
+        boolean open = false;
+
+        try {
+            open = database.isOpen();
+        } finally {
+            // releasing read lock
+            lock.readLock().unlock();
+            return open;
+        }
+    }
+
+    @Override
+    public boolean isClosed() {
+        // isOpen also handles locking
+        return !isOpen();
+    }
+
+    @Override
+    public boolean isLocked() {
+        // being able to acquire a write lock means that the resource is not locked
+        // only one write lock can be taken at a time, also excluding any concurrent read locks
+        if (lock.writeLock().tryLock()) {
+            lock.writeLock().unlock();
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    @Override
+    public boolean isAutoCommitEnabled() {
+        // no locks because the autocommit flag never changes
+        return database.isAutoCommitEnabled();
+    }
+
+    @Override
+    public boolean isPersistent() {
+        // no locks because the persistence flag never changes
+        return database.isPersistent();
+    }
+
+    @Override
+    public boolean isCreatedOnDisk() {
+        // acquire read lock
+        lock.readLock().lock();
```

Comment:
Can be simplified to:

~~~~java
lock();
try {
  return database.isCreatedOnDisk();
} finally {
  unlock();
}
~~~~